build.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- ============================================================== -->
+<!-- Phing Build instructions                                       -->
+<!-- http://www.phing.info/                                         -->
+<!-- http://www.phing.info/get/phing-latest.phar                    -->
+<!-- php -d phar.readonly=Off phing-latest.phar phar-build-release  -->
+<!-- ============================================================== -->
+<project name="roave-backward-compatibility-check" default="phar-build">
+    <property name="build-dir" value="build"/>
+    <property name="phar-dir" value="build/phar"/>
+
+    <!-- ============================================  -->
+    <!-- Target: phar-prepare-dependencies             -->
+    <!-- ============================================  -->
+    <target name="phar-prepare-dependencies">
+        <!--install dependencies without development requirements-->
+        <exec command="composer install --no-dev -o" passthru="true" checkreturn="true"/>
+    </target>
+
+    <!-- ============================================  -->
+    <!-- Target: phar-build                            -->
+    <!-- ============================================  -->
+    <target name="phar-build" depends="phar-prepare-dependencies">
+        <!--create the package-->
+        <mkdir dir="${build-dir}" />
+        <php expression="file_put_contents('bin/clistub.php', '#!/usr/bin/env php' . chr(10) . Phar::createDefaultStub('bin/roave-backward-compatibility-check.php'))"/>
+        <pharpackage basedir="./"
```

Comment:
We really shouldn't drag in phing for just this 😬

Reply:
This is just based off the suggested approach in maglnet/ComposerRequireChecker - which uses Phing's PHAR packager to make this heh

Reply:
Yup - I think we just need to extract this specific step to be done via bash, and then we are good 🤔

Reply:
Seems like it isn't too much Voodoo: https://github.com/phingofficial/phing/blob/d98239f270e86c6a886eb93893d5ddd17c9d71bd/src/Phing/Task/System/PharPackageTask.php

Only dependency becomes ext-phar in require-dev

Reply:
Sorry for bothering: any update on this? Thanks!

Reply:
/cc @asgrim ofr this: I completely dropped the ball on PHARs on my end.

Reply:
I'm not sure how else to do this? The phing approach gets it done, but you seem hesitant to bring in the dep... Don't know about other approaches at this time.

Reply:
Hmm, yeah, looking at https://github.com/maglnet/ComposerRequireChecker/blob/fb2a69aa2b7307541233536f179275e99451b339/.github/workflows/phar-creation.yml#L32, there's no real way around it.

We could have a `tools/phing/composer.json`, but let's roll with the current solution for now, if you are happy with it.

I can extract it from here later, if it becomes a problem: would really rather not have phing in my big list of problems, but it is what it is :D

Reply:
Not sure if @theseer might be working a tool: https://phpc.social/@theseer/109296480712703364

Reply:
Let's just go with what works now and unblocks @sebastianbergmann, and then we suffer later in trying to extract it from our pipeline again :P

Reply:
> Not sure if @theseer might be working a tool: [phpc.social/@theseer/109296480712703364](https://phpc.social/@theseer/109296480712703364)

This is work in progress and in early design phase on my side.
Given I'm currently quite busy, this is likely not to materialize very soon :-(